data/models/noteModel.js

```diff
@@ -0,0 +1,138 @@
+const db = require('../dbConfig.js');
+
+const get = async (id) => {
+    if (id) {
+        const note = await db('notes').where('id', id);
+        if (note.length) {
+            return note[0];
+        } else {
+            const e = new Error("id does not exist");
+            e.name = "InvalidID";
+            throw e;
+        }
+    }
+
+    const notes = await db('notes');
+    return notes;
+};
+
+const insert = async (note) => {
+
+    // check for missing note object
+    if (typeof note === 'undefined') {
+        const e = new Error("note object");
+        e.name = "MissingParam";
+        throw e;
+    }
+
+    // check if note is object
+    if (typeof note !== 'object') {
+        const e = new TypeError("note is not an object");
+        throw e;
+    }
+
+    // check for missing title key
+    if (!note.hasOwnProperty('title')) {
```

Comment:
Since `note` isn't inherited, you are fine with just checking `!note.title`

Side Note: Now if it something like 
```
const originalNote = { title: 'Here is a title', content: 'Some content' }

const note = originalNote;

console.log(note.hasOwnProperty('title')) // false because it is inherited

note.tag = 'here is a tag'

console.log(note.hasOwnProperty('tag)) // true
```